src/types.js

```diff
@@ -341,30 +341,30 @@ export type UnreadStreamItem = {|
   |}>,
 |};
 
-export type TimePieceDescriptor = {|
+export type MessageListElementTime = {|
   type: 'time',
   key: string,
   timestamp: number,
   subsequentMessage: Message | Outbox,
 |};
 
-export type MessagePieceDescriptor = {|
+export type MessageListElementMessage = {|
   type: 'message',
   key: number,
   message: Message | Outbox,
   isBrief: boolean,
 |};
 
-export type HeaderPieceDescriptor = {|
+export type MessageListElementHeader = {|
   type: 'header',
   key: string,
   subsequentMessage: Message | Outbox,
 |};
 
-export type HtmlPieceDescriptor =
-  | TimePieceDescriptor
-  | MessagePieceDescriptor
-  | HeaderPieceDescriptor;
+export type MessageListElement =
+  | MessageListElementTime
+  | MessageListElementMessage
+  | MessageListElementHeader;
```

Comment:
I keep wanting to read these type names as some kind of time, some kind of message, etc. How about `TimeMessageListElement` and so on?